blog/2017-12-07-streaming-to-elasticsearch.adoc

```diff
@@ -0,0 +1,255 @@
+= Streaming Data Changes from Your Database to Elasticsearch
+jpechane
+:awestruct-tags: [ mysql, postgres, elasticsearch, smt, example ]
+:awestruct-layout: blog-post
+
+We have seen how to set up a change data stream to a downstream database link:/blog/2017/09/25/streaming-to-another-database[few weeks ago].
+
+In this blog post we will follow the same approach to stream the data to an https://www.elastic.co/[Elasticsearch] server for potential further search processing. But to make the matter a little bit more interesting, we will stream the data to both, a PostgreSQL database and Elasticsearch, so we will optimize access to the data via the SQL query language as well as via full-text search.
+
+== Topology
+The general topology for this scenario is shown in the following picture:
+
+.A General topology
+[#img-general]
+[caption="Figure 1: "]
+image::dbz-to-multiple.svg[Scenario topology]
+
+&nbsp; +
+
+We are going to simplify the deployment again, so we will use only one Kafka Connect instance that will contain all connectors and will serve as both as entry point and exit point for Kafka.
+
+&nbsp; +
+
+.A Simplified topology
+[#img-general]
+[caption="Figure 2: "]
+image::dbz-to-multiple-simplified.svg[Scenario topology]
+
+== Configuration
+
+We will use this https://github.com/debezium/debezium-examples/tree/master/unwrap-smt[Docker Compose file] for a fast deployment of the demo.
+The deployment consists of following Docker images:
+
+* https://hub.docker.com/r/debezium/zookeeper/[Apache ZooKeeper]
+* https://hub.docker.com/r/debezium/kafka/[Apache Kafka]
+* An https://github.com/debezium/debezium-examples/tree/master/unwrap-smt/debezium-jdbc[enriched] Kafka Connect / Debezium https://hub.docker.com/r/debezium/connect/[image] with changes
+** PostgreSQL JDBC driver placed into `/kafka/libs` directory
+** https://docs.confluent.io/current/connect/connect-jdbc/docs/index.html[Kafka Connect JDBC Connector] (developed by https://www.confluent.io/[Confluent]) placed into `/kafka/connect/kafka-connect-jdbc` directory
+* Pre-populated MySQL used in our link:docs/tutorial[tutorial]
+* Empty PostgreSQL
+* Empty ElasticSearch
+
+We again use the `UnwrapFromEnvelope` transformation to automatically extract a new row record and thus effectively convert the Debezium change data message into a simple one consumable by other connectors. Both JDBC and Elasticsearch connectors expect the same format.
```

Comment:
This (using "again") makes a bit less sense if someone hasn't read the previous blog post.

Also, it's not as clear as it could be why this is even needed. Somewhere, you should probably explain that the connectors were developed separately from each other and chose to represent the events a bit differently. Debezium uses a more complex event structure so that it captures all of the information available. This `UnwrapFromEnvelope` single message transformation (SMT) collapses this complex change event structure into the same row-based format expected by the two sink connectors.